src/extensions/deflate.rs

```diff
@@ -0,0 +1,922 @@
+//! Permessage-deflate extension
```

Comment:
Did you write this file manually? Or is it a modified copy-paste from `deflate`? I'm asking, because it's quite a huge file and I and @agalakhov will have to validate it carefully as it contains `loop`s, `unwrap`s and some copy-pasting.

Reply:
It's an adaptation of WS-RS's deflate extension to use ZLib instead of libc, and to work with a growable fragment buffer.

Reply:
Got it. I think I or @agalakhov will have to take a close look at it. At the first glance, it seems like it contains some copy-paste that we would like to avoid :)